### DISCUSSION TITLE
Don't build sphinx documentation

### DISCUSSION THREAD

**Discussion by spvkgn** [kovidgoyal/kitty#3967] [Q&A]
Hi, is there an option to disable building sphinx docs with `python3 setup.py linux-package` ?

**Comment by kovidgoyal** [comment]
On Sun, Aug 22, 2021 at 04:39:43AM -0700, spvkgn wrote:
> Hi, is there an option to disable building sphinx docs?

Read the Makefile, it has lots of targets, choose the one you want

**Reply by derekschrock** [reply]
I think the issue here is building of docs is forced by `create_macos_bundle_gunk` and `create_linux_bundle_gunk`  can this be disabled?

**Reply by page-down** [reply]
@derekschrock

Recent discussion (See comments section):
https://github.com/kovidgoyal/kitty/commit/3288400005041d6963e06cf6dd1187c115b1046f

> sphinx will be a build-time dependency to run kitty.app from source code and is no longer optional

Based on the setup.py code, I currently run `mkdir -p docs/_build/{man,html}` once if I don't want to install sphinx within some environment.
Or modify the code, there is currently no command line option to skip this step.

I believe that sphinx should be available as an optional dependency when kitty is compiled from source and run, since there is a website available to access the documentation.
However, kovid believes that the documentation only needs to be compiled once, not slowing down each build, and that the filesystem IO caused by copying these documentation files each time is irrelevant.

**Reply by kovidgoyal** [reply]
You don't need to create bundles other than on macOS because of Apple limitations with non-bundled binaries. Just building from source on linux doesnt require sphinx. Although I want the documentation to be available locally not just online and having different capabilities in different builds just makes everything harder.
